### PR TITLE
Update error messages for questions for DOS5 services

### PR DESCRIPTION
**Pull request by lfdebrux**
Ticket: https://trello.com/c/Bt72BXJZ/430-update-error-messages-for-dos-5

I used the [Design System guidance](https://design-system.service.gov.uk/components/error-message/) on writing error messages to structure these.

I've grouped questions with similar error messages together to make reviewing easier and also to make it easier to make changes from reviews.

### CODE REVIEW

File: frameworks/digital-outcomes-and-specialists-5/questions/services/agileCoachLocations.yml [hunk 1]
```diff
@@ -24,4 +24,4 @@ options:
 
 validations:
   - name: answer_required
-    message: 'You need to answer this question.'
+    message: Select the location or locations
```
Comment (bjgill) [inline frameworks/digital-outcomes-and-specialists-5/questions/services/agileCoachLocations.yml:27]: Can we make this shorter? The GOV.UK style guide seems to suggest [just using the plural](https://www.gov.uk/guidance/style-guide/a-to-z-of-gov-uk-style#brackets)

(and all other 'X or Xs' in messages)

File: frameworks/digital-outcomes-and-specialists-5/questions/services/anonymousRecruitment.yml [hunk 2]
```diff
@@ -1,5 +1,6 @@
 name: Provide discreet recruitment
-question: Can you, if asked, recruit participants without revealing details of the organisation you’re recruiting for?
+question: >
+  Can you, if asked, recruit participants without revealing details of the organisation you’re recruiting for?
```
Comment (bjgill) [inline frameworks/digital-outcomes-and-specialists-5/questions/services/anonymousRecruitment.yml:3]: Why this change?

File: frameworks/digital-outcomes-and-specialists-5/questions/services/anonymousRecruitment.yml [hunk 3]
```diff
@@ -8,4 +9,6 @@ depends:
 
 validations:
   - name: answer_required
-    message: 'You need to answer this question.'
+    message: >
+      Select yes if you can provide recuit participants without revealing
```
Comment (bjgill) [inline frameworks/digital-outcomes-and-specialists-5/questions/services/anonymousRecruitment.yml:13]:
```suggestion
      Select if you can recruit participants without revealing
```

^ typo, and I think the 'yes' is superfluous (compare to https://github.com/alphagov/digitalmarketplace-frameworks/blob/0cf756cfe473b7c8fd7b98bd44fa5197a8cbd19b/frameworks/g-cloud-12/questions/services/serviceAddOnType.yml#L25) (and also other 'Select yes if...')

**Review by bjgill — COMMENTED** [review]
A significant improvement. I've got a question and a couple of content suggestions.

File: frameworks/digital-outcomes-and-specialists-5/questions/services/agileCoachLocations.yml (hunk 1 above)
Comment (lfdebrux) [reply frameworks/digital-outcomes-and-specialists-5/questions/services/agileCoachLocations.yml:27]: It's based on [Design System guidance on errors for checkboxes](https://design-system.service.gov.uk/components/checkboxes/#error-messages).

I'm happy to change it but I'd want to see what Nora thinks first.

File: frameworks/digital-outcomes-and-specialists-5/questions/services/anonymousRecruitment.yml (hunk 2 above)
Comment (lfdebrux) [reply frameworks/digital-outcomes-and-specialists-5/questions/services/anonymousRecruitment.yml:3]: Woops, didn't mean to do that

File: frameworks/digital-outcomes-and-specialists-5/questions/services/anonymousRecruitment.yml (hunk 3 above)
Comment (lfdebrux) [reply frameworks/digital-outcomes-and-specialists-5/questions/services/anonymousRecruitment.yml:13]: It's based off the [Design System guidance on error messages for radios](https://design-system.service.gov.uk/components/radios/).

Again I'm happy to change it, if Nora thinks it's a good idea, but in general we should be sticking to the Design System patterns where possible.

Comment (bjgill) [reply frameworks/digital-outcomes-and-specialists-5/questions/services/anonymousRecruitment.yml:13]: OK. That makes sense. I clearly need to get better acquainted with the Design System guidance - I'm far more familiar with the GOV.UK guidance, which can be subtly different.

Was this intentional, though?

```suggestion
      Select yes if you can recruit participants without revealing
```

Comment (lfdebrux) [reply frameworks/digital-outcomes-and-specialists-5/questions/services/anonymousRecruitment.yml:13]: > Was this intentional, though?

Nope, good spot :D

**Review by bjgill — APPROVED** [review]
:rocket:

**Review by NoraGDS — APPROVED** [review]

File: frameworks/digital-outcomes-and-specialists-5/questions/services/recruitFromList.yml [hunk 4]
```diff
@@ -8,4 +8,6 @@ type: boolean
 
 validations:
   - name: answer_required
-    message: 'You need to answer this question.'
+    message: >
+      Select yes if you are willing to recruit participants based on a list
+      provided you you by the buyer
```
Comment (NoraGDS) [inline frameworks/digital-outcomes-and-specialists-5/questions/services/recruitFromList.yml:13]:
```suggestion
      provided to you by the buyer
```

**Review by NoraGDS — CHANGES_REQUESTED** [review]

File: frameworks/digital-outcomes-and-specialists-5/questions/services/recruitFromList.yml (hunk 4 above)
Comment (lfdebrux) [reply frameworks/digital-outcomes-and-specialists-5/questions/services/recruitFromList.yml:13]: Woops, thanks.

**Review by NoraGDS — APPROVED** [review]

**Review by NoraGDS — APPROVED** [review]

**Review by NoraGDS — APPROVED** [review]

**Review by NoraGDS — APPROVED** [review]

File: frameworks/digital-outcomes-and-specialists-5/questions/services/softwareDevelopmentTypes.yml [hunk 5]
```diff
@@ -40,4 +40,4 @@ options:
 
 validations:
   - name: answer_required
-    message: 'You need to answer this question.'
+    message: Select the type or types software development you provide
```
Comment (NoraGDS) [inline frameworks/digital-outcomes-and-specialists-5/questions/services/softwareDevelopmentTypes.yml:43]:
```suggestion
    message: Select the type or types of software development you can provide
```

**Review by NoraGDS — CHANGES_REQUESTED** [review]

**Review by NoraGDS — APPROVED** [review]

File: frameworks/digital-outcomes-and-specialists-5/questions/services/softwareDevelopmentTypes.yml (hunk 5 above)
Comment (lfdebrux) [reply frameworks/digital-outcomes-and-specialists-5/questions/services/softwareDevelopmentTypes.yml:43]: Good spot

**Review by NoraGDS — APPROVED** [review]

**Review by NoraGDS — APPROVED** [review]

**Review by NoraGDS — APPROVED** [review]
